sources/osgGA/CADManipulatorHammerController.js

```diff
@@ -0,0 +1,323 @@
+        this._zoomFactor = 5.0;
+
+        this._lastScale = 0;
+        this._nbPointerLast = 0; // to check if we the number of pointers has changed
+
+        this._dimensionMask = ( 1 << 2 );
+        this._lastPos = Vec2.create(); // to set the pivot for rotation
+    },
+    setEventProxy: function ( proxy ) {
+        if ( proxy === undefined || ( proxy !== undefined && proxy === this._eventProxy ) ) {
+            return;
+        }
+        this._eventProxy = proxy;
+        var self = this;
+        var hammer = proxy;
+        var computeTouches = function ( event ) {
+            if ( event.pointers !== undefined )
+                return event.pointers.length;
+            return 1; // mouse
+        };
+
+        var dragCB = function ( ev ) {
+            return 'touches ' + computeTouches( ev ) + ' distance ' + ev.distance + ' x ' + ev.deltaX + ' y ' + ev.deltaY;
+        };
+        // Set a minimal thresold on pinch event, to be detected after pan
+        hammer.get( 'pinch' ).set( {
+            threshold: 0.1
+        } );
+        // Let the pan be detected with two fingers.
+        hammer.get( 'pan' ).set( {
+            threshold: 0,
+            pointers: 0
+        } );
+        hammer.get( 'pinch' ).recognizeWith( hammer.get( 'pan' ) );
+
+        hammer.get( 'tap' ).set( {
+            taps: 2,
+            posThreshold: 300
+        } );
+
+        this._cbPanStart = ( function () {
+            var origIntersect = Vec3.create();
+            var dstIntersect = Vec3.create();
+
+            return function ( event ) {
+                var manipulator = self._manipulator;
+                if ( !manipulator || self._transformStarted || event.pointerType === 'mouse' ) {
+                    return;
+                }
+                var gesture = event;
+                self._dragStarted = true;
+                self._nbPointerLast = computeTouches( gesture );
+
+                var pos;
+                if ( self._nbPointerLast === 2 ) {
+                    pos = self.getPositionRelativeToCanvas( gesture );
+                    self._lastPos = pos;
+                } else {
+                    if ( self._lastPos.length === 0 ) {
+                        pos = self.getCanvasCenter( gesture );
+                    } else {
+                        pos = self._lastPos;
+                    }
+                }
+
+                var hits = [];
+                var cam = manipulator.getCamera();
+                var point, matrix, pTrans;
+
+                var width = cam.getViewport().width();
+                var height = cam.getViewport().height();
+                manipulator.getRotateInterpolator().setWidth( width );
+                manipulator.getRotateInterpolator().setHeight( height );
+                manipulator.getPanInterpolator().setWidth( width );
+                manipulator.getPanInterpolator().setHeight( height );
+
+                var iv = manipulator.getIntersectionVisitor();
```

Comment:
Move the getIntersection function from CADManipulatorStandardMouseKeyboard to CADManipulator, so we can use it here too.